packages/shared/src/contexts/NotificationsContext.tsx

```diff
@@ -0,0 +1,195 @@
+import React, {
+  ReactElement,
+  ReactNode,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  UseNotificationPermissionPopup,
+  useNotificationPermissionPopup,
+} from '../hooks/useNotificationPermissionPopup';
+import { BootApp } from '../lib/boot';
+import { isDevelopment, isTesting } from '../lib/constants';
+import AuthContext from './AuthContext';
+
+interface NotificationsContextData extends UseNotificationPermissionPopup {
+  unreadCount: number;
+  isInitialized: boolean;
+  isSubscribed: boolean;
+  isNotificationSupported: boolean;
+  isNotificationsReady?: boolean;
+  clearUnreadCount: () => void;
+  incrementUnreadCount: () => void;
+  onTogglePermission: () => Promise<NotificationPermission>;
+}
+
+const NotificationsContext =
+  React.createContext<NotificationsContextData>(null);
+
+export default NotificationsContext;
+
+export interface NotificationsContextProviderProps {
+  children: ReactNode;
+  unreadCount?: number;
+  isNotificationsReady?: boolean;
+  app: BootApp;
+}
+
+export const NotificationsContextProvider = ({
+  app,
+  children,
+  isNotificationsReady,
+  unreadCount = 0,
+}: NotificationsContextProviderProps): ReactElement => {
+  const isExtension = !!process.env.TARGET_BROWSER;
```

Comment:
Is see we use this quite often actually, perhaps it's best to utilize this somewhere as one function.
If it would ever change we can easily manage it.

Reply:
I agree. A much better procedure.